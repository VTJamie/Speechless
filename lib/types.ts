export abstract class ModelProvider<T> {
  abstract model(): T;
  abstract toJSON(): T;
}

export type Calculable<A> = A | Expression;

export class Expression {
  private readonly expression: string;
  private constructor(expression: string) {
    this.expression = expression;
  }
  static val(expression: string) {
    return new Expression(expression);
  }

  toJSON(): string {
    return this.expression;
  }
}

export const e = (val: string): Expression => {
  return Expression.val(val);
};

export type LocaleString = 'en-US' | 'en-CA' | 'en-GB';

export type SpeechType = 'PlainText' | 'SSML';
