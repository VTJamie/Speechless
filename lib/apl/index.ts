export * from './commands';
export * from './components';
export * from './document';
export * from './filters';
export * from './interfaces';
export * from './avg';
export * from './gestures';
