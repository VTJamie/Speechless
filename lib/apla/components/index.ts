export * from './component';
export * from './audio';
export * from './mixer';
export * from './selector';
export * from './sequencer';
export * from './silence';
export * from './speech';
