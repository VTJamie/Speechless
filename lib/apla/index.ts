export * from './document';
export * from './interfaces';
export * from './components';
