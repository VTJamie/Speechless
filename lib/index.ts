export * as skill from './skill';
export * as apl from './apl';
export * as apla from './apla';
// export * as acdl from './acdl';
export * from './types';
