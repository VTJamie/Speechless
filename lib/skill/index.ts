export * from './intent';
export * from './locale';
export * from './models';
export * from './skills';
export * from './types';
