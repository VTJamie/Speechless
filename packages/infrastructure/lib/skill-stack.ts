import { Stack, StackProps, Duration } from 'aws-cdk-lib';
import { Construct } from 'constructs';

export interface SkillStackProps extends StackProps {  
}

export class SkillStack extends Stack {
  constructor(scope: Construct, id: string, props: SkillStackProps) {
    super(scope, id, props);        
  }
}
