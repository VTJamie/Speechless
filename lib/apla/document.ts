import { interfaces } from 'ask-sdk-model';
import * as fs from 'fs';
import * as uuid from 'uuid';

import { APLABaseComponentModel, APLABaseComponentProps, APLAComponent } from './components';
import { APLAResourcesProps } from './interfaces';
import { ModelProvider } from '../';

export interface APLADocumentModel {
  type?: string;
  version?: string;
  description?: string;
  // compositions?: CompositionProps[];
  resources?: APLAResourcesProps[];
  mainTemplate: {
    parameters: string[];
    items: APLAComponent<APLABaseComponentModel, APLABaseComponentProps>[];
  };
}

export interface APLADocumentProps {
  documentName: string;
  description?: string;
  // compositions?: CompositionProps[];
  resources?: APLAResourcesProps[];
  items: APLAComponent<APLABaseComponentModel, APLABaseComponentProps>[];
}

export class APLADocument implements ModelProvider<APLADocumentModel> {
  readonly props: APLADocumentProps;
  constructor(props: APLADocumentProps) {
    this.props = props;
  }

  getDirective(additionalData?: unknown): interfaces.alexa.presentation.apla.RenderDocumentDirective {
    const data = additionalData;
    return {
      type: 'Alexa.Presentation.APLA.RenderDocument',
      token: uuid.v4(),
      document: {
        src: `doc://alexa/apla/documents/${this.props.documentName}`,
        type: 'Link'
      },
      datasources: {
        ...(data ? data : {})
      }
    };
  }

  getFullDirective(additionalData?: unknown): interfaces.alexa.presentation.apla.RenderDocumentDirective {
    const data = additionalData;
    return {
      type: 'Alexa.Presentation.APLA.RenderDocument',
      token: uuid.v4(),
      document: {
        ...this.model()
      },
      datasources: {
        ...(data ? data : {})
      }
    };
  }

  toJSON(): APLADocumentModel {
    return this.model();
  }

  model(): APLADocumentModel {
    return {
      type: 'APLA',
      version: '0.91',
      description: this.props.description,
      mainTemplate: {
        parameters: [],
        items: this.props.items
      }
    };
  }

  writeToFile(baseSkillPackagePath?: string) {
    const aplaDocumentFolder = (baseSkillPackagePath ? baseSkillPackagePath : './') + `response/prompts/${this.props.documentName}`;
    if (!fs.existsSync(aplaDocumentFolder)) {
      fs.mkdirSync(aplaDocumentFolder, { recursive: true });
    }

    const model = this.model();
    // write interaction models
    fs.writeFileSync(`${aplaDocumentFolder}/document.json`, JSON.stringify(model, null, 2));
  }
}
