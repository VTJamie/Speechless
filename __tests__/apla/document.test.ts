import { APLADocument } from '../../lib/apla';
import { matchSnapshot } from '../helper';

matchSnapshot(
  new APLADocument({
    documentName: 'test-apla',
    items: []
  })
);
