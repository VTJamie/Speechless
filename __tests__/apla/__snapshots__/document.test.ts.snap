// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APLADocument: APLADocument 1`] = `
{
  "mainTemplate": {
    "items": [],
    "parameters": [],
  },
  "type": "APLA",
  "version": "0.91",
}
`;
